### ISSUE TITLE
Hololens needs Game Controller Support in Unity

### ISSUE THREAD

**Issue by Subere23** [Microsoft/HoloToolkit-Unity#97]
There does not seem to be any support for game controllers as of yet. Bluetooth controllers will pair with the Hololens but do not function inside of Unity apps. The same controller will work paired to the computer inside the editor or standalone version of the game, which leads me here for solutions.

**Comment by Brekel** [comment]
Well there's this: http://brekel.com/gamepad-serverclient/

**Comment by SUBERE23** [comment]
Its a nice idea, but unfortunately that tethers a Hololens to a computer. Kind of defeats one of Hololen's main design features.

**Comment by NeerajW** [comment]
@Subere23 would the clicker solve the scenarios you are looking for?

**Comment by Subere23** [comment]
Maybe for one of my pet projects that would suffice, but not the other one.
Two of my projects I have tried using Gaze, Gesture and Voice to control
but they need a more detailed and much faster interactions. The idea of
mixing a controller with Gaze and voice seems really appealing as well. In
the end though, if having a game controller working on Hololens is too far
outside of the set paradigm, which I can get an argument for, I can explore
farther in other directions. The response time on gesture and voice is
rather slow. It is a bit faster with the clicker.

That said, I can always use a bluetooth keyboard and mouse, if I cant find
better solutions. Though that solution feels more tethered.

On Wed, Jul 6, 2016 at 10:52 AM, Neeraj Wadhwa <notifications@github.com>
wrote:

> @Subere23 <https://github.com/Subere23> would the clicker solve the
> scenarios you are looking for?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/HoloToolkit-Unity/issues/97#issuecomment-230815839>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG3e5ceuJe5VAT2OaJ1VmxWz-OZQX_7Xks5qS89HgaJpZM4JFofe>
> .
>



-- 
Jason Odom
*Thought Experiment*
205-223-1011
Jasonodom23@gmail.com

**Comment by flarb** [comment]
I agree 100% -- we need controller support. I seem to remember the HoloLens twitter account tweeted me about this and suggested that HoloLens will support the upcoming Xbox One S controller since it has bluetooth?

**Comment by Subere23** [comment]
Unfortunately that is at the end of August if that is the case.


On Wed, Jul 6, 2016 at 11:34 AM, Ralph Barbagallo <notifications@github.com>
wrote:

> I agree 100% -- we need controller support. I seem to remember the
> HoloLens twitter account tweeted me about this and suggested that HoloLens
> will support the upcoming Xbox One S controller since it has bluetooth?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/HoloToolkit-Unity/issues/97#issuecomment-230829512>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG3e5dD4vf0RK8RO07HVDsmC1Rf-SGU-ks5qS9kKgaJpZM4JFofe>
> .
>



-- 
Jason Odom
*Thought Experiment*
205-223-1011
Jasonodom23@gmail.com

**Comment by keveleigh** [comment]
The most recent Unity drop (I'm seeing it in b22) added Bluetooth to the Capabilities under Publishing Settings. Have you tried your controller with this option checked?

**Comment by Subere23** [comment]
Bluetooth has been there, afaik, since the first build of HTP that I had in April. I always check the bluetooth option, in the hopes that my bluetooth controller would start working, and also I have the bluetooth keyboard and mouse. Of course, the only times I have ever seen a controller working with the Hololens is Mr. Brekel's demo and the E3 Minecraft demo from 2015.

**Comment by riverar** [comment]
The Xbox One S controller is out August 2 (and may already in stock @ Building 92). Not a long wait. Are you not getting any controller events? Have you tried adding that HoloLens Input component (built into Unity)? I haven't had time to dig into that yet but it does light up some keyboard input scenarios.

**Comment by jwittner** [comment]
Is this a HoloToolkit-Unity bug or a Unity bug? We have projects in the works to bring Game Controller support to HoloToolkit-Unity before Unity would get it for UWP? Seems ambitious, but we'd be pioneers.

**Comment by Subere23** [comment]
Josh, I have been going back and forth with Unity support to get
controllers working with UWP. A few days ago it looks like we got this
issue fixed. Hopefully we will find out shortly...

The last post from that thread..

"Thanks, that explains the problem, on UWP Unity was only connecting to
joysticks (with UsagePage 0x01 and Usage 0x04 as per 'Special purpose
hardware ID' table
https://msdn.microsoft.com/en-us/library/windows/hardware/ff538842(v=vs.85).aspx
)

But it seems the devices you're using, are classified as gamepads -
UsagePage 0x01 and Usage 0x05

I fixed the problem in our main branch, and will graft the fixes to
upcoming Unity 5.3 and 5.4.

Sorry for the inconvience"
